ADR/0006-log-conventions.md

```diff
@@ -11,39 +12,36 @@ Accepted
 
-In our controller logs, we will use structured log messages, partially formatted in JSON with key-value pairs as described below.
+## Decision: Log conventions
 
-```
-TIMESTAMP LOG_LEVEL LOGGER MESSAGE ADDITIONAL_JSON_DATA
-```
+In our controller logs, we will use structured log messages, formatted in JSON with key-value pairs as described below.
 
 For example:
 
 ```
-2023-03-07T11:32:29.167Z    INFO    controllers.Component   updating devfile component name kubernetes-deploy ...   {"namespace": "samburrai-tenant", "resource": "go-net-http-hello-prqx", "kind": "Component"}
-2023-03-07T11:32:29.020Z    INFO    controllers.Component   API Resource updated {"namespace": "samburrai-tenant", "audit": "true", "resource": "go-net-http-hello-prqx", "kind": "Component", "action": "UPDATE"}
+{"ts": "2023-03-07T11:32:29.167Z", "level": "info", "logger": "controllers.Component", "msg": "updating devfile component name kubernetes-deploy ...", "namespace": "samburrai-tenant", "name": "go-net-http-hello-prqx", "controllerKind": "Component"}
+{"ts": "2023-03-07T11:32:29.020Z", "level": "info", "logger": "controllers.Component", "msg": "API Resource updated", "namespace": "samburrai-tenant", "audit": "true", "name": "go-net-http-hello-prqx", "controllerKind": "Component", "action": "UPDATE"}
 ```
 
-The meaning of **timestamp** and log level should be self evident. The **logger** name helps your team understand which part of your controller is emitting the log (usually `Log.WithName` from `zap`). The **message** is a human readable string describing the event being logged. The **json data** contains additional fields useful for searching.
+The **ts** field stands for "timestamp" (encoded as UTC/ISO-8601) and the meaning of log **level** should be self evident. The **logger** name helps your team understand which part of your service is emitting the log (usually `Log.WithName` from `zap`). The **msg** is a human readable string describing the event being logged. Further **json** key value pairs contain additional fields useful for searching.
 
 The opinionated logging mechanism from the controller-runtime framework already conforms to this scheme. Care should be taken to make sure specific key/value pairs are logged.
 
 ### 1. When did it happen?
 
-**Included in:** `TIMESTAMP`
+**Included in:** `ts`
 
-Encode timestamps in UTC/ISO-8601 format at the start of the log line.
+Encode timestamps in UTC/ISO-8601 format.
 
 ### 2. What happened?
 
-**Included in:** `MESSAGE`, `ADDITIONAL_JSON_DATA`
+**Included in:** `msg` and `action`
 
 Use the key `action` with possible values `VIEW`, `ADD`, `UPDATE`, and `DELETE`, if applicable.
```

Comment:
Why does `CREATE` not included?

Reply:
Maybe we can use standard CRUD (Create, Read, Update, Delete)? Or kubernetes verbs (delete, create, get, update/patch). Is it `VIEW`, `ADD`, `UPDATE`, and `DELETE` some standard, it slightly invokes SQL (probably due to upper case) but there is INSERT  and SELECT.

Reply:
No, I think the `VIEW`, `ADD`, `UPDATE`, and `DELETE` are not based on some standard. We have the freedom to change it here, so long as we agree and settle on something.

Any strong opinions on this? If we're going to change it, let's change it now and let this settle.

Reply:
Well, I had use-case where creation of an object failed. So, I had to log the error with action CREATE which is not listed here.

Reply:
Would `ADD` not be acceptable in that situation?